frontend/src/components/material/useMaterialViewHelper.js

```diff
@@ -0,0 +1,141 @@
+import { computed, onMounted, ref } from 'vue'
+import { useRoute } from 'vue-router/composables'
+import * as XLSX from 'xlsx'
+import { slugify } from '@/plugins/slugify.js'
+import i18n from '@/plugins/i18n/index.js'
+import dayjs from '@/common/helpers/dayjs.js'
+import { apiStore } from '@/plugins/store/index.js'
+import { materialFromRoute } from '@/router.js'
+
+function generateFilename(camp, materialList) {
+  const filename = materialList
+    ? [
+        slugify(camp.name),
+        slugify(i18n.tc('components.material.useMaterialViewHelper.detail')),
+        slugify(materialList),
+      ]
+    : [
+        slugify(camp.name),
+        slugify(i18n.tc('components.material.useMaterialViewHelper.overview')),
+      ]
+  return [...filename, dayjs().format('YYMMDDHHmmss')].join('_') + '.xlsx'
+}
+
+async function getActivity(camp, materialItem) {
```

Comment:
here it bites us that we don't have a reference from ContentNode to activity.
The same functionality is already in MaterialTable::methods::findOneActivityByContentNode.
Maybe the code can be shared by using this method in MaterialTable::methods::findOneActivityByContentNode.

Reply:
@BacLuc could you create an issue for this?